scripts/jointLklDM.C

```diff
@@ -451,20 +485,64 @@ void jointLklDM(TString configFileName="$GLIKESYS/rcfiles/jointLklDM.rc",Int_t s
+                      {
+                        dNdEFileName[iChannel] = fdNdEDir+"dNdESignal_"+channelval[iChannel]+Form("_%.1fmass.root",mass);
+                        cout << "      " << iChannel+1 << ") " << dNdEFileName[iChannel] << " (with BR_" << iChannel+1 << " = " << brval[iChannel] << ") ... " << endl;
+                      }
+                    TString* dNdEFileNamePointer = dNdEFileName;
+                    // Try to read dNdE from existing files
+                    if(fullLkl->ReaddNdESignal(nChannels,dNdEFileNamePointer,brval))
+                      {
+                        cout << "Failed! <---------------- FATAL ERROR!!!" << endl;
+                        return;
+                      }
+                    else
+                      cout << "Ok!" << endl;
+                  }
+                else if(!channel.CompareTo("branon",TString::kIgnoreCase))
+                  {
+                    //
+                    // This is a placeholder for the function that calculate the BR in the branon model
+                    //
+                    // The only input of this function is the dark matter mass ('mass' at this point in gLike)
+                    //
+                    TString  dNdEFileName[nChannels];
+                    for(UInt_t iChannel=0;iChannel<nChannels;iChannel++)
+                      {
+                        dNdEFileName[iChannel] = fdNdEDir+"dNdESignal_"+channelval[iChannel]+Form("_%.1fmass.root",mass);
+                        cout << "      " << iChannel+1 << ") " << dNdEFileName[iChannel] << " (with BR_" << iChannel+1 << " = " << brval[iChannel] << ") ... " << endl;
+                      }
+                    TString* dNdEFileNamePointer = dNdEFileName;
+                    // Try to read dNdE from existing file
+                    if(fullLkl->ReaddNdESignal(nChannels,dNdEFileNamePointer,brval))
+                      {
+                        cout << "Failed! <---------------- FATAL ERROR!!!" << endl;
+                        return;
+                      }
+                    else
+                      cout << "Ok!" << endl;
+                  }
+                else
+                  {
+                    TString dNdESignalFileNameForm   = fdNdEDir+"dNdESignal_"+channel+Form("_%smass.root",mprecform.Data());
+                    const TString dNdESignalFileName = Form(dNdESignalFileNameForm,(isDecay? mass/2. : mass));
+                    cout << "   * Reading dN/dE for signal from file " << dNdESignalFileName  << "... " << flush;
+
+                    // Try to read dNdE from existing file
+                    if(fullLkl->ReaddNdESignal(dNdESignalFileName))
+                      {
+                        cout << "Failed! <---------------- FATAL ERROR!!!" << endl;
+                        return;
+                      }
+                    else
+                      cout << "Ok!" << endl;
+                  }
```

Comment:
Also all this part needs to be redone to adapt to my first comment of this review